sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/ControlFlowGraph/ControlFlowGraphBuilder.cs

```diff
@@ -479,16 +479,27 @@ private void BuildLabeledStatement(LabeledStatementSyntax labeledStatement)
 
         private void BuildTryStatement(TryStatementSyntax tryStatement)
         {
+            var finallyBlock = currentBlock;
             if (tryStatement.Finally?.Block != null)
             {
-                currentBlock = CreateBlock(currentBlock);
+                finallyBlock = currentBlock = CreateBlock(currentBlock);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of 'currentBlock' from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1121)